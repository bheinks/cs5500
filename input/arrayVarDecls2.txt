(*
	Declarations of arrays with more than one
      variable per declaration
*)

program p;

var A : array [1..5] of integer;
    B,C : array [20..50] of boolean;
    D,E,F : array [100..500] of char;

begin
  write(5)
end.

