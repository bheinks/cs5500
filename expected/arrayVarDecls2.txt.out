

>>> Entering new scope...

+++ Adding p to symbol table with type PROGRAM

+++ Adding A to symbol table with type ARRAY 1 .. 5 OF INTEGER

+++ Adding B to symbol table with type ARRAY 20 .. 50 OF BOOLEAN

+++ Adding C to symbol table with type ARRAY 20 .. 50 OF BOOLEAN

+++ Adding D to symbol table with type ARRAY 100 .. 500 OF CHAR

+++ Adding E to symbol table with type ARRAY 100 .. 500 OF CHAR

+++ Adding F to symbol table with type ARRAY 100 .. 500 OF CHAR

<<< Exiting scope...

---- Completed parsing ----
